pkg/common/cns-lib/node/manager.go

```diff
@@ -185,6 +189,16 @@ func (m *defaultManager) GetNodeNameByUUID(ctx context.Context, nodeUUID string)
 	return nodeName, nil
 }
 
+// GetK8sNode returns Kubernetes Node object for the given node name
+func (m *defaultManager) GetK8sNode(ctx context.Context, nodename string) (*v1.Node, error) {
+	log := logger.GetLogger(ctx)
+	node, err := m.k8sClient.CoreV1().Nodes().Get(ctx, nodename, metav1.GetOptions{})
+	if err != nil {
+		log.Errorf("failed to obtain node for nodename:%q", nodename)
```

Comment:
return error here ?

Reply:
we are returning an error right after printing this log.